app/src/renderer/components/wallet/SendModal.vue

```diff
@@ -205,45 +140,38 @@ export default {
   },
   mounted() {
     if (this.denom) {
-      this.fields.denom = this.denom
+      this.denom = this.denom
     }
   },
   methods: {
     ...mapActions([`sendTx`]),
-    close() {
-      this.$emit(`update:showSendModal`, false)
+    open(denom) {
+      this.denom = denom
+      this.$refs.actionModal.open()
     },
-    async validateForm() {
-      this.sending = true
+    validateForm() {
       this.$v.$touch()
 
-      if (!this.$v.$invalid) {
-        await this.submitForm()
-      } else {
-        this.sending = false
-      }
+      return !this.$v.$invalid
     },
-    async submitForm() {
-      const amount = +this.fields.amount
-      const address = this.fields.address
-      const denom = this.fields.denom
+    async submitForm(submitType, password) {
+      const amount = +this.amount
+      const address = this.address
+      const denom = this.denom
       const type = `send`
 
-      await this.$refs.actionModal.submit(async () => {
-        await this.sendTx({
-          type,
-          password: this.fields.password,
-          to: address,
-          amount: [{ denom, amount: amount.toString() }]
-        })
-
-        this.$store.commit(`notify`, {
-          title: `Successfully Sent`,
-          body: `Successfully sent ${amount} ${denom} to ${address}`
-        })
-      }, `Sending tokens failed`)
+      await this.sendTx({
+        type,
+        submitType,
+        password: password,
+        to: address,
+        amount: [{ denom, amount: amount.toString() }]
```

Comment:
I'm a little obsessive with `String()` over `.toString()` because the latter fails if `amount` is `undefined`
```suggestion
        amount: [{ denom, amount: String(amount) }]
```